### ISSUE TITLE
setting and reading custom options

### ISSUE THREAD

**Issue by bumberboy** [connectrpc/connect-go#600]
Hello there

Thanks for building this!

I'm looking to set some custom options on my rpcs:

```
import "google/protobuf/descriptor.proto";

extend google.protobuf.MethodOptions {
  optional string abc = 51234;
}

service XXXService {
  rpc Xxx(XxReq) returns (XxResp) {
    option (abc) = "hello, world";
  }

}

```

And be able to read them within an interceptor. The intention is to build an authorization interceptor that checks the custom option on the rpc on request to run the right ACL auth validation. Any idea how I can achieve this?

Thanks in advance!

**Comment by emcfarlane** [comment]
As an example it might be good to have a look at: https://github.com/akshayjshah/connectauth

Can use context to propagate auth values.

**Comment by bumberboy** [comment]
@emcfarlane Thanks for the suggestion! That looks perfect for authentication (and I've already implemented something similar to that.

I'm looking to implement authorization now, where I try to answer the question: can this user perform this action (READ, WRITE, DELETE) on this resource?

I'm intending to annotate each rpc that requires authorization with a `resource` and an `action`, and then within my authorization interceptor, make a call to an external service that determines if the user has the right permissions. I'm looking to use spicedb from authzed for the auth service.

**Comment by jeffmccune** [comment]
I'm working on a similar use case Ed asked me to provide feedback on.  I'd like to include primitive roles (owner, editor, viewer) to start my project and check "has rbac" off the list, then potentially add fine grained permissions later if customers need it.

I've got an ID token with a groups claim, which is a json array of strings.  Groups in this specific case is really a role and may be a different claim depending on the id provider.  Keycloak for example has a role claim.

It would buy a lot of runway early in the project if I could tag each service method as allowing messages from one of owners, editors, or viewers.

One specific use case I've been thinking of:  Given a PatchUser(UserRequest) UserResponse method, allow only requests with user.email == idToken.email so that users cannot set their email to arbitrary values.

**Comment by jhump** [comment]
This should be addressed by #629.
